cmd/dao_tests/resource_test.go

```diff
@@ -0,0 +1,165 @@
+//go:build integration
+// +build integration
+
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	_ "github.com/RHEnVision/provisioning-backend/internal/dao/sqlx"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func createPkResource() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeNoop,
+		Handle:   "handle1",
+	}
+}
+
+func createPkResource2() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeAzure,
+		Handle:   "handle1",
+	}
+}
+
+func setupResource(t *testing.T) (dao.PubkeyResourceDao, context.Context) {
+	setup()
+	ctx := identity.WithTenant(t, context.Background())
+	resourceDao, err := dao.GetPubkeyResourceDao(ctx)
+	if err != nil {
+		panic(err)
+	}
+	return resourceDao, ctx
+}
+
+func teardownPubkeyResource(_ *testing.T) {
+	teardown()
+}
+
+func TestCreateResource(t *testing.T) {
+	resourceDao, ctx := setupResource(t)
+	defer teardownPubkeyResource(t)
+	resource := createPkResource()
+	err := resourceDao.Create(ctx, resource)
+	if err != nil {
+		panic(err)
+	}
+	resources, err := resourceDao.ListByPubkeyId(ctx, resource.PubkeyID)
+	if err != nil {
+		panic(err)
+	}
+
+	assert.Equal(t, 1, len(resources))
+	assert.Equal(t, resource.ID, resources[0].ID)
+	assert.Equal(t, resource.Provider, resources[0].Provider)
+	assert.Equal(t, resource.Handle, resources[0].Handle)
+	assert.Equal(t, resource.Tag, resources[0].Tag)
```

Comment:
When you comparing the whole struct, then you can take advantage of Go equality contract: two structs are equal if all fields are equal:

```suggestion
	assert.Equal(t, resource, resources[0])
```